src/modules/ModuleRunner.ts

```diff
@@ -17,31 +17,142 @@ limitations under the License.
+        // Set up defaults
+        this._cryptoSetup = new DefaultCryptoSetupExtensions();
+        this._experimental = new DefaultExperimentalExtensions();
+    }
+
+    /**
+     * Provides a crypto setup extension.
+     *
+     * @returns The registered extension. If no module provides this extension, a default implementation is returned
+     */
+    public get cryptoSetup(): ProvideCryptoSetupExtensions {
+        return this._cryptoSetup;
+    }
+
+    /**
+     * Provides a n experimental extension.
+     *
+     * @remarks
+     * This method extension is provided to simplify experimentaion an development, and is not intended for production code
+     *
+     * @returns The registered extension. If no module provides this extension, a default implementation is returned
+     */
+    public get experimental(): ProvideExperimentalExtensions {
+        return this._experimental;
+    }
+
+    /**
+     * Resets the extension to the defaults
+     *
+     * Intended for test usage only.
+     */
+    public reset(): void {
+        this._implementionMap = new ExtensionImplementationMap();
+        this._cryptoSetup = new DefaultCryptoSetupExtensions();
+        this._experimental = new DefaultExperimentalExtensions();
+    }
+
+    /**
+     * Add any extensions provided by the module
+     *
+     * @param module - The appModule to check for extensions
+     *
+     * @throws if an extension is provided by more than one module
+     *
+     */
+    public addExtensions(module: AppModule): void {
+        const runtimeModule = module.module;
+
+        /* Record the cryptoSetup extensions if any */
+        if (runtimeModule.extensions?.cryptoSetup) {
+            if (this._implementionMap.hasDefaultCryptoSetupExtension) {
+                this._cryptoSetup = runtimeModule.extensions?.cryptoSetup;
+                this._implementionMap.hasDefaultCryptoSetupExtension = false;
+            } else {
+                throw new Error(
+                    `adding cryptoSetup extension implementation from module ${runtimeModule.moduleName} but an implementation was already provided`,
+                );
+            }
+        }
+
+        /* Record the experimental extensions if any */
+        if (runtimeModule.extensions?.experimental) {
+            if (this._implementionMap.hasDefaultExperimentalExtension) {
+                this._experimental = runtimeModule.extensions?.experimental;
+                this._implementionMap.hasDefaultExperimentalExtension = false;
+            } else {
+                throw new Error(
+                    `adding experimental extension implementation from module ${runtimeModule.moduleName} but an implementation was already provided`,
+                );
+            }
+        }
+    }
+}
+
 /**
  * Handles and coordinates the operation of modules.
  */
 export class ModuleRunner {
     public static readonly instance = new ModuleRunner();
 
+    private _extensions = new ExtensionsManager();
+
```

Comment:
No need to use a `_` prefix on private properties -- typescript enforces that for us.

```suggestion
    private extensions = new ExtensionsManager();
```

Reply:
This was to distinguish between the internal private property from the publicly exposed get method.

I will change names to avoid confusion with JS "private" naming convention `_`

Reply:
> This was to distinguish between the internal private property from the publicly exposed get method.

Oh, this makes sense. Sorry, that is a sensible thing to do!

renaming it to `extensionsManager` is fine as well.